Homework-№4-from-28.09.2022/task 2/main.c

```diff
@@ -0,0 +1,30 @@
+#include "task2.h"
+
+int main()
+{
+    tests();
+    char fileName[100] = {"\0"};
+    printf("Hello! Please, enter the filename with an extension on your computer\n");
+    printf("in this folder, which contains a natural number N and after N integer numbers\n");
+    printf("I will find most popular element in this array faster, than O(n^2)!\n");
+    printf("Your filename =");
+    scanf("%s", fileName);
+    FILE *file = fopen(fileName, "r");
```

Comment:
```suggestion
    FILE *file = fopen(fileName, "r");
    if (file == NULL)
    {
         printf(...);
         return -2;
    }
```
Нет проверки, что файл успешно открыт.
Может, файл не существует? Или у запущенной программы нет доступа к файлу?